torch/_dynamo/variables/tensor.py

```diff
@@ -71,17 +71,18 @@ def get_real_value(self):
     def __init__(
         self,
         proxy: torch.fx.Proxy,
-        dtype=None,
-        device=None,
-        layout=None,
-        ndim=None,
+        *,
```

Comment:
It's a little brittle to have this list:

1) here
2) in `def specialize` as

```
            "dtype": value.dtype,
            "device": value.device,
            "layout": value.layout,
            "ndim": int(value.ndim),
            "requires_grad": value.requires_grad,
            "is_quantized": value.is_quantized,
            "is_sparse": value.is_sparse,
            "class_type": type(value),
```

3) in `var_getattr` for access as 

```
if name == "$name" and self.$name is not None:
            result = ConstantVariable(self.$name, **options)
```
(sometimes Constant is another cls, like shape)

4) Vaguely overlapping w/ `_nonvar_fields`

What can we do to make it easier to map the relationships here, and make it easier to see when one misses a step if adding a new specialized var (For example, adding it to the specialize list but not the ctor makes a weird VT error iirc, or adding it to the two but not the getattr makes for an incorrect fallthrough)

Reply:
I can do a bigger refactor after this. The point of the change here is it establishes a new invariant (which is important, because I don't want people to break the invariant).